data/transactions/logic/eval.go

```diff
@@ -610,6 +620,9 @@ func EvalContract(program []byte, gi int, aid basics.AppIndex, params *EvalParam
 	if params.Ledger == nil {
 		return false, nil, errors.New("no ledger in contract eval")
 	}
+	if params.SigLedger == nil {
+		params.SigLedger = params.Ledger
```

Comment:
Is ledger copying okay in terms of performance in LogicSigs? Also should something similar be done in `EvalSignature` to initialize the `SigLedger` there?

Reply:
However, a final thing to work out is whether real ledger and the sig ledger, as I've passed them in, agree on what `Round()` is. I think the SigLedger is giving the last committed round, and the Ledger is giving the one being worked on. So I have to track that detail down and ensure the opcodes work the same in sigs and apps.

Reply:
Thanks - I'll resolve as the `SigLedger` check in `EvalSignature` looks good to me